### PR TITLE
Publishers from api

### PR DESCRIPTION
**Pull request by StRhys**
this pr is a number of tickets rolled into one.

This adds api calls to get:
- homepage: publishers + link to catalogue filtered
- homepage: topics + link to topic page
- homepage: dataset count
- topics: latest datasets + link to dataset
- topics: all datasets filtered link
- topics: subtopics + link to catalogue filtered
- topics: publishers + link to catalogue filtered

**to test**
have a play around, click through the topics and publishers, sanity check code.
Some of code around the filtering might look slightly dodge, this is just to get it filtering correctly

### CODE REVIEW

File: components/CardList/index.tsx [hunk 1]
```diff
@@ -29,23 +31,85 @@ const CardListPublisherCard = ({ items }: { items: CardProps[] }) => {
...
+  );
+};
+
+const CardListTopicCard = ({
+  items,
+}: {
+  items: { title: string; description: string; "@id": string }[];
+}) => {
+  const getTopicHref = (id: string) => {
+    const splitTopic = id.split("/");
```
Comment (mikeAdamss) [inline components/CardList/index.tsx]: if I'm reading this right you can get what you want from the "identifier" field, have corrected the data now (in pr as a write this) so it should work (and will always be the case - we just need more data validation).

File: libs/dataRequests.tsx [hunk 2]
```diff
@@ -16,9 +16,11 @@ const getHeaders = () => {
 };
 
 const handleResponse = async (response: Response) => {
-  if (!response.ok) {
-    throw new Error(`Failed to fetch data. Status: ${response.status}`);
-  }
+  // TODO rework how we handle responses
+  // commented outt for the time being as was throwing errors on no subtopics
```
Comment (mikeAdamss) [inline libs/dataRequests.tsx:20]: when you get to this, look for a 404, that's our standard response on the api for empty thingy.

File: components/FilterSelection/index.tsx [hunk 3]
```diff
@@ -162,7 +162,7 @@ const topics = [
 const publishers = [
   { heading: "All publishers" },
   {
-    heading: "Department for Business, Energy & Industrial Strategy",
+    heading: "The Department for Business, Energy & Industrial Strategy",
```
Comment (mikeAdamss) [inline components/FilterSelection/index.tsx]: Gareth made me take all the "The " prefixes out of the api the other day, we should probably be consistent.

Wider point, we've got a getPublishers fetch around here somewhere, can we just fetch these?

File: components/DatasetsList/index.tsx [hunk 4]
```diff
@@ -179,20 +179,34 @@ export default function DatasetsList({
...
+          x.topics.some((topic: string) =>
+            topic.includes(activeTopicFilter.toLowerCase().replaceAll(" ", "-"))
+          )
         );
       }
 
       if (activeSubtopicsFilter?.length > 0) {
         filteredData = filteredData.filter((x) =>
-          activeSubtopicsFilter.includes(x.subTopic)
+          x.topics.some((topic: string) => {
```
Comment (mikeAdamss) [inline components/DatasetsList/index.tsx:191]: if I'm understanding, this is recreating the "identifier" from the "title" of the topic?

not end of the world and we can manage (especially with data validation) but there's probably a world where they don't quite match so we'll need to look at this after the demo.

**Review by mikeAdamss — COMMENTED** [review]
Couple of comments and questions and one issue.

If I click through "It and internet. industry" the page tells me 0 datasets for that topic - but there's 1.

In this instance it's the 2nd of 2 topics for the 4gc dataset, not sure if its connected but fyi.

File: components/FilterSelection/index.tsx (hunk 3 above)
Comment (StRhys) [reply components/FilterSelection/index.tsx]: I think this got pulled in when we was making some changes.
there's a ticket in this sprint for it, hooking these filter selections up to the api is next on my todo list

File: components/DatasetsList/index.tsx (hunk 4 above)
Comment (StRhys) [reply components/DatasetsList/index.tsx:191]: I did have a worry that could be the case, my backup idea was slightly longer but it's getting each topic and working backwards from that. Which will be the safer solution in the future.

**Review by mikeAdamss — DISMISSED** [review]
LGTM

**Review by mikeAdamss — APPROVED** [review]

**Pull request merged**